### ISSUE TITLE
Repo: Set no-else-return's allowElseIf: false internally

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [typescript-eslint/typescript-eslint#10278]
### Suggestion

Following #6822 -> #7340: we have https://eslint.org/docs/latest/rules/no-else-return enabled internally, but don't configure it. It's [`allowElseIf` option](https://eslint.org/docs/latest/rules/no-else-return#options) is `true` by default to be more lenient.

Proposal: let's make it stricter internally?

### Additional Info

Inspired by https://github.com/typescript-eslint/typescript-eslint/pull/10194/files#diff-3cc43b506517aba406ae3a31d7998b116c4b09db76b632c6cb3c2e54fc71d66cR77. cc @kirkwaiblinger 

💖

**Comment by kirkwaiblinger** [comment]
Booooo 🤣

For the record, I strongly disagree with the likes of no-else-return and no-lonely-if, and I am deeply entrenched in this opinion 🫣

However, I do think the rule makes more sense with `allowElseIf` disabled, so, granted we are no-else-return users, I suppose this makes sense.

**Comment by Zamiell** [comment]
Why don't you like no-else-return / no-lonely-if?

**Comment by kirkwaiblinger** [comment]
TL;DR I abhor nontrivial mutually dependent conditions in many situations.

Personally, I think of 
```ts
if (conditionA) {
} else if (conditionB) {
} else {
}
```
as something that can always be safely reordered to 
```ts
if (conditionB) {
} else if (conditionA) {
} else {
}
```
regardless of whether there are control flow statements in the bodies.

If I mean something that cannot be reordered, I write
```ts
if (conditionA) {
} else {
  if (conditionB) {
  } else {
  }
} 
```
so that it _can_ be safely reordered as 
```ts
if (!conditionA) {
  if (conditionB) {
  } else {
  }
} else {
}
```

Both no-else-return and no-lonely-if (especially the even stricter unicorn/no-lonely-if) intefere with this approach.

**Comment by peanutenthusiast** [comment]
I started working on this issue, so this helps. Thanks Kirk.

**Comment by Zamiell** [comment]
Personally, I use switch statements be for "logic that can be reordered", and I use if statements to be for "logic that can't be reordered". Or, put differently, the algorithm would be something like: "Always use switch statements, and only use if statements if representing the logic as a switch statement would be impossible." Is there a problem with that approach? Do you just never use switch statements?

**Comment by kirkwaiblinger** [comment]
> Personally, I use switch statements for "logic that can be reordered", and if statements to be for "logic that can't be reordered". Or, put differently, the algorithm would be something like: "Always use switch statements, and only use if statements if representing the logic as a switch statement would be impossible." Is there a problem with that approach? Do you just never use switch statements?

Heh, well don't get me started on the people that use `switch` statements in a particular order for their fallthrough behavior 😆, see https://github.com/typescript-eslint/typescript-eslint/issues/10241, https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#an_alternative_to_if...else_chains.

At the end of the day, no, there's not a problem with your approach; it's completely valid, too. _My_ mental model favors treating branches of `if`s and `switch`es as "equally important" in many cases, and, therefore, not using "guard clause" style as an ordinary branching mechanism. Somehow this all probably can be romanticized as a great "declarative"/"functional" vs "imperative" mindset, or as something to do with envisioning control flow as a directed graph with branches to many nodes at once, rather than sequentially splitting in two... 🤷. But, it doesn't really matter. Your mind and everyone else's are essentially a black box to me, and all that matters is that we can communicate effectively and collaborate on working code, not that we have the same mental pictures thereof 🙂. I was watching a Technology Connections video one day, and the presenter dropped the immensely profound line ["the only thing better than perfect is standardized"](https://micro.denisthiessen.de/the-only-thing-better-than-perfect-is-standardized), and I find it extremely applicable here. While I'd rather we all used my preferred style, which I do strongly believe is better for reading, writing, _and_ editing, it's even better that we converge on the majority consensus.
